distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Base_Generator.enso

```diff
@@ -214,14 +214,26 @@ make_is_in arguments = case arguments.length of
     0 -> Error.throw <| Illegal_State_Error_Data ("The operation IS_IN requires at least one argument.")
     ## If only the self argument is provided, no value will ever be in the empty list, so we just short circuit to false.
        `IN ()` would be more meaningful, but it is a syntax error.
-    1 -> code '2=1' . paren
+    1 -> code 'FALSE' . paren
```

Comment:
Fair, I was thinking of postponing that until we actually implement such a backend that does not support it, but maybe better to prepare for it already.

Reply:
Okay, now I remember the reason I did it this way - currently we don't have a way from inside a function generating the SQL code to call-back into the dialect. The idea is that these methods are self-contained.

Once SQL Server is implemented it can just provide an override of the whole `IS_IN` operation and use a different constant there.

Do you think we should allow to call-back into the dialect from within these functions? That should indeed be possible and it could allow for some more modularity in constructing these expressions - for example we could transform a single expression into a whole stack of expressions and then generate code for that. This would allow to re-use some parts.
On the other hand, it will make the 'recipes' translating an IR expression into SQL less direct - currently it is quite clear what is the resulting SQL code just looking at the recipe and that is beneficial IMO.

Reply:
Our tests should detect if the current implementation does not work for a new Dialect and we can then override it accordingly in the new dialect. So we can keep the current default for now.